core/tests/protractor_utils/ProfilePage.js

```diff
@@ -41,18 +41,26 @@ var ProfilePage = function() {
   };
 
   this.expectCurrUserToHaveProfilePhoto = async function() {
+    await waitFor.visibilityOf(currUserProfilePhoto,
+      'Current user profile picture label element taking too long to appear');
     expect(await currUserProfilePhoto.isPresent()).toBe(true);
   };
 
   this.expectOtherUserToHaveProfilePhoto = async function() {
+    await waitFor.visibilityOf(otherUserProfilePhoto,
+      'otherUserProfilePhoto label element taking too long to appear');
     expect(await otherUserProfilePhoto.isPresent()).toBe(true);
   };
 
   this.expectUserToHaveBio = async function(expectedText) {
+    await waitFor.visibilityOf(bio,
+      'bio label element taking too long to appear');
     expect(await bio.getText()).toMatch(expectedText);
   };
 
   this.expectUserToHaveNoInterests = async function() {
+    await waitFor.visibilityOf(interests,
+      'interests label element taking too long to appear'); //Err
     var numInterests = await interests.count();
     expect(numInterests).toEqual(0);
   };
```

Comment:
For `expectUserToHaveInterests`, if `expectedInterests` isn't empty, you can wait for `interests.first()` to be visible